File: archived/tests/transpiler/x/php/outer_join.php
<?php
$customers = [["id" => 1, "name" => "Alice"], ["id" => 2, "name" => "Bob"], ["id" => 3, "name" => "Charlie"], ["id" => 4, "name" => "Diana"]];
$orders = [["id" => 100, "customerId" => 1, "total" => 250], ["id" => 101, "customerId" => 2, "total" => 125], ["id" => 102, "customerId" => 1, "total" => 300], ["id" => 103, "customerId" => 5, "total" => 80]];
$result = (function() use ($customers, $orders) {
  $result = [];
  foreach ($orders as $o) {
    $matched = false;
    foreach ($customers as $c) {
      if (!($o["customerId"] == $c["id"])) continue;
      $matched = true;
      $result[] = ["order" => $o, "customer" => $c];
    }
    if (!$matched) {
      $c = null;
      $result[] = ["order" => $o, "customer" => $c];
    }
  }
  foreach ($customers as $c) {
    $matched = false;
    foreach ($orders as $o) {
      if (!($o["customerId"] == $c["id"])) continue;
      $matched = true;
      break;
    }
    if (!$matched) {
      $o = null;
      $result[] = ["order" => $o, "customer" => $c];
    }
  }
  return $result;
})();
echo "--- Outer Join using syntax ---", PHP_EOL;
foreach ($result as $row) {
  if ($row["order"]) {
  if ($row["customer"]) {
  echo "Order" . " " . (is_float($row["order"]["id"]) ? json_encode($row["order"]["id"], 1344) : $row["order"]["id"]) . " " . "by" . " " . (is_float($row["customer"]["name"]) ? json_encode($row["customer"]["name"], 1344) : $row["customer"]["name"]) . " " . "- $" . " " . (is_float($row["order"]["total"]) ? json_encode($row["order"]["total"], 1344) : $row["order"]["total"]), PHP_EOL;
} else {
  echo "Order" . " " . (is_float($row["order"]["id"]) ? json_encode($row["order"]["id"], 1344) : $row["order"]["id"]) . " " . "by" . " " . "Unknown" . " " . "- $" . " " . (is_float($row["order"]["total"]) ? json_encode($row["order"]["total"], 1344) : $row["order"]["total"]), PHP_EOL;
};
} else {
  echo "Customer" . " " . (is_float($row["customer"]["name"]) ? json_encode($row["customer"]["name"], 1344) : $row["customer"]["name"]) . " " . "has no orders", PHP_EOL;
}
}

File: archived/tests/transpiler/x/java/outer_join.java
public class Main {
    static Data1[] customers = new Data1[]{new Data1(1, "Alice"), new Data1(2, "Bob"), new Data1(3, "Charlie"), new Data1(4, "Diana")};
    static class Data1 {
        int id;
        String name;
        Data1(int id, String name) {
            this.id = id;
            this.name = name;
        }
        boolean containsKey(String k) {
            if (k.equals("id")) return true;
            if (k.equals("name")) return true;
            return false;
        }
    }

    static Data2[] orders = new Data2[]{new Data2(100, 1, 250), new Data2(101, 2, 125), new Data2(102, 1, 300), new Data2(103, 5, 80)};
    static class Data2 {
        int id;
        int customerId;
        int total;
        Data2(int id, int customerId, int total) {
            this.id = id;
            this.customerId = customerId;
            this.total = total;
        }
        boolean containsKey(String k) {
            if (k.equals("id")) return true;
            if (k.equals("customerId")) return true;
            if (k.equals("total")) return true;
            return false;
        }
    }

    static java.util.List<Result4> result = new java.util.ArrayList<Result4>() {{ java.util.ArrayList<Result4> _tmp = new java.util.ArrayList<>(); for (var o : orders) { for (var c : customers) { if (((Integer) (o.get("customerId"))) == c.id) { _tmp.add(new Result4(o, c)); } } } java.util.ArrayList<Result4> list = _tmp; int skip = 0; int take = -1; for (int i = 0; i < list.size(); i++) { if (i < skip) continue; if (take >= 0 && i >= skip + take) break; _tmp.add((Result4)list.get(i)); } addAll(_tmp);}};
    static class Result4 {
        Data2 order;
        Object customer;
        Result4(Data2 order, Object customer) {
            this.order = order;
            this.customer = customer;
        }
        boolean containsKey(String k) {
            if (k.equals("order")) return true;
            if (k.equals("customer")) return true;
            return false;
        }
    }


    public static void main(String[] args) {
        System.out.println("--- Outer Join using syntax ---");
        for (var row : result) {
            if (row.order) {
                if (row.customer) {
                    System.out.println("Order" + " " + row.order.id + " " + "by" + " " + row.customer.name + " " + "- $" + " " + row.order.total);
                } else {
                    System.out.println("Order" + " " + row.order.id + " " + "by" + " " + "Unknown" + " " + "- $" + " " + row.order.total);
                }
            } else {
                System.out.println("Customer" + " " + row.customer.name + " " + "has no orders");
            }
        }
    }
}
